internal/ui/desk.go

```diff
@@ -96,86 +63,130 @@ func (l *deskLayout) MinSize(objects []fyne.CanvasObject) fyne.Size {
 	return fyne.NewSize(640, 480) // tiny - the window manager will scale up to screen size
 }
 
-func (l *deskLayout) newDesktopWindow() fyne.Window {
+func (l *deskLayout) newDesktopWindow(outputName string) fyne.Window {
 	if l.wm == nil {
-		win := l.app.NewWindow("Fyne Desktop (Embedded)")
+		win := l.app.NewWindow("Embedded " + RootWindowName)
 		win.SetPadded(false)
 		return win
 	}
 
-	desk := l.app.NewWindow("Fyne Desktop")
+	desk := l.app.NewWindow(fmt.Sprintf("%s%s", RootWindowName, outputName))
 	desk.SetPadded(false)
-	desk.FullScreen()
+	desk.SetFullScreen(true)
 
 	return desk
 }
 
 func (l *deskLayout) updateBackgrounds(path string) {
-	for _, background := range l.backgrounds {
-		background.updateBackgroundPath(path)
-		canvas.Refresh(background)
+	for bg := range l.backgroundScreenMap {
+		bg.updateBackgroundPath(path)
+		canvas.Refresh(bg)
 	}
 }
 
-func (l *deskLayout) Root() fyne.Window {
-	if l.win != nil {
-		return l.win
-	}
-
-	l.win = l.newDesktopWindow()
-	if l.win.Canvas().Scale() != l.screens.Scale() {
-		l.win.Canvas().SetScale(l.screens.Scale())
+func (l *deskLayout) createRoot(screen *desktop.Screen) {
+	win := l.newDesktopWindow(screen.Name)
+	l.roots = append(l.roots, win)
+	bg := newBackground()
+	l.backgroundScreenMap[bg] = screen
+	if screen == l.screens.Primary() {
```

Comment:
No it isn't a duplicate - one is refactoring an already existing window and widget panel/bar - the other is creating a window/widget panel/bar.

Reply:
I don't mean it's not running in the same context, I mean the lines of code are literally the same.
``` go
	l.bar = newBar(l)
	l.widgets = newWidgetPanel(l)
	l.mouse = newMouse()
	l.mouse.Hide() // temporarily we do not draw mouse (using X default)
```

Reply:
Actually they aren't duplicates - the duplication is already factored out in createRoot() - the first code creates a primary screen and the second code is simply there to refactor a screen that already exists into a root screen rather than create one.

Reply:
Oh okay I see now - you weren't referring to the above code but the widget code - gotcha